bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/client/Client.java

```diff
@@ -62,7 +63,12 @@ public class Client {
 
     public Client(HttpClient httpClient) {
         this.httpClient = httpClient;
-        userAgent = "openHAB/" + FrameworkUtil.getBundle(this.getClass()).getVersion().toString();
+        String bundleVersion = "unknown";
+        Bundle bundle = FrameworkUtil.getBundle(this.getClass());
+        if (bundle != null) {
+            bundleVersion = bundle.getHeaders().get("Bundle-Version");
+        }
+        userAgent = "openHAB-EntsoeBinding/" + bundleVersion;
```

Comment:
Can be simplified to:
```suggestion
        userAgent = "openHAB-EntsoeBinding/" + org.openhab.core.OpenHAB.getVersion();
```

Is there any specific reason for adding the "-EntsoeBinding" postfix?

Reply:
When using `org.openhab.core.OpenHAB.getVersion()` there is no need for null checks.

Reply:
Nah, still needed for unit tests. If code is exectued outside OSGi there's no bundle and execution fails. That's why I keep the bundle null check

<img width="691" height="715" alt="image" src="https://github.com/user-attachments/assets/1486561c-2b66-4820-a9d9-cde956f4a5e6" />

Reply:
An easy solution might be something like this:

https://github.com/openhab/openhab-addons/blob/89b92263378eb848d8251e61845ec15d8cc01828/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/ApiController.java#L92

https://github.com/openhab/openhab-addons/blob/89b92263378eb848d8251e61845ec15d8cc01828/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/ApiController.java#L97

https://github.com/openhab/openhab-addons/blob/89b92263378eb848d8251e61845ec15d8cc01828/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/ApiController.java#L122

It only defers initialization though, so if actual HTTP calls are being made, it won't solve anything.

Reply:
Sorry for the late reply.

In general production code shouldn't "know" when it's being run from a unit test, or behave differently. Normally such cases can be resolved by DI and/or interfaces.

As an example for this case, the `EnsoeClient` constructor could have a parameter taking a `VersionProvider` interface with one method: `getVersion`. One implementation would simply return `org.openhab.core.OpenHAB.getVersion()` for this method, while the unit test implementation would return "unknown" (like in your current version).

It may feel like too much for this small case, and I agree it's only a minor issue. So if @lsiepel agrees, we can accept it as is, but perhaps you could then add a small comment explaining why it's implemented like that with such a "fallback" that seems unneeded/strange at first glance?